MRI_geom/SPD_fMRI.py

```diff
@@ -0,0 +1,158 @@
+import networkx as nx
+import numpy as np
+import matplotlib.pyplot as plt
+import pandas as pd
+import time
+from sklearn.svm import SVC
+from sklearn.model_selection import KFold
+from sklearn.metrics import precision_score, accuracy_score
+from sklearn.metrics import f1_score, recall_score, confusion_matrix
+import geomstats.spd_matrices_space as spd_space
+SPACE = spd_space.SPDMatricesSpace(dimension=28)
+
+
+graphs = pd.DataFrame.from_csv("data/train_FNC.csv")
+map_FCN = pd.DataFrame.from_csv("AdditionalInformation/comp_ind_fMRI.csv",
+                                index_col=None)
+map_FCN = map_FCN['fMRI_comp_ind'].to_dict()
+map_FCN_r = {v: k for k, v in map_FCN.iteritems()}
+mapping = pd.DataFrame.from_csv("AdditionalInformation/rs_fMRI_FNC_mapping.csv")
+graph_labels = pd.DataFrame.from_csv(
+                                     "data/train_labels.csv")
+all_graphs = [None]*graphs.shape[0]
+all_targets = [None]*graphs.shape[0]
+vpos = np.vectorize(lambda x: max(0, x))
+for i in range(graphs.shape[0]):
+    u = np.zeros((28, 28))
+    for ii in range(mapping.shape[0]):
+        a, b = (mapping.iloc[ii]["mapA"], mapping.iloc[ii]["mapB"])
+        u[map_FCN_r[a], map_FCN_r[b]] = graphs.iloc[i][ii]
+    u = np.multiply(u, (u > 0.0))
+    all_graphs[i] = nx.from_numpy_array(u + u.T)
+    all_targets[i] = graph_labels.loc[graphs.index[i], "Class"]
+
+
+np.random.seed(2018)
+index_train = range(len(all_graphs))
+np.random.shuffle(index_train)
+T = int(0.85*len(all_graphs))
+labels = np.array([all_targets[t] for t in index_train])
+
+
+def Laplacian(A):
```

Comment:
no uppercases.

Reply:
A -> `adjacency_mat` is more self-explanatory. We should understand most of the code without looking at the paper.